src/runtime/HalideRuntime.h

```diff
@@ -541,6 +546,38 @@ extern int halide_copy_to_host(void *user_context, struct halide_buffer_t *buf);
 extern int halide_copy_to_device(void *user_context, struct halide_buffer_t *buf,
                                  const struct halide_device_interface_t *device_interface);
 
+/** Copy data from one buffer to another. The buffers may have
+ * different shapes and sizes, but the destination buffer's shape must
```

Comment:
Might be good to be clear here that only the mins and extents count. I assume the strides can be arbitrarily different.